### PR TITLE
JUnit failures counter

### PR DESCRIPTION
**Pull request by SteinHeselmans**
Counts number of failures in a JUnit xml file

note: the counting using regex is a bit different from the one above, as multiple failures can be on one line in a single line xml-file. We might want to change the regex statement for doxy/sphinx as well?

note: also made the --doxygen --sphinx options non exclusive... (elif --> if)

### CODE REVIEW

File: src/mlx/warnings.py [hunk 1]
```diff
@@ -39,12 +42,23 @@ def return_doxygen_warnings(self):
         print("{count} doxygen warnings found".format(count=self.doxygen_counter))
         return self.doxygen_counter
 
+    def check_junit_failures(self, line):
+        '''
+        Function for counting the number of JUnit warnings in a xmlfile.
+        The function returns the number of failing test cases found
+        '''
+        self.junit_counter += len(re.findall(junit_pattern, line))
```
Comment (bavovanachte) [inline src/mlx/warnings.py]: This only gives us a count of the warnings, correct? Do we want to log the names of the failed testcases as well?

Comment (SteinHeselmans) [inline src/mlx/warnings.py]: yes, this only is a counter. I didn't print the warnings/failures as the 2 other ones also don't print their  warnings. This makes sense as the parsing is from a terminal output which usually appears right on before the warnings plugin is invoked, so in CI you'll see one on top of the other. For JUnit we're parsing xml which is not displayed in the terminal.

Maybe we could add a feature request, where we have a command argument to generically enable/disable output detailed warning information per case ?

**Comment by Letme** [comment]
:+1:

File: src/mlx/warnings.py (hunk 1 above)
Comment (bavovanachte) [inline src/mlx/warnings.py]: :+1:

Comment (SteinHeselmans) [inline src/mlx/warnings.py]: should be covered in #19

**Pull request merged**